framework/include/meshgenerators/Poly2TriMeshGenerator.h

```diff
@@ -0,0 +1,54 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MeshGenerator.h"
+
+/**
+ * Generates a triangulation in the XY plane, based on an input mesh
+ * defining the outer boundary and an optional set of input meshes
+ * defining inner hole boundaries.
+ */
+class Poly2TriMeshGenerator : public MeshGenerator
+{
+public:
+  static InputParameters validParams();
+
+  Poly2TriMeshGenerator(const InputParameters & parameters);
+
+  std::unique_ptr<MeshBase> generate() override;
+
+protected:
```

Comment:
Mind adding some doxygen comments in here? I also think that most of this can be const

Reply:
Done